File: problemas/1100/1117.php
<?php

$nota1 = readline();

while ($nota1 < 0 || $nota1 > 10) {
    echo "nota invalida\n";
    $nota1 = readline();
}

$nota2 = readline();

while ($nota2 < 0 || $nota2 > 10) {
    echo "nota invalida\n";
    $nota2 = readline();
}

echo 'media = ' . number_format(($nota1 + $nota2) / 2, 2) . "\n";

File: problemas/1100/1117.cs
using System;

namespace uri1117
{
    internal static class Program
    {
        private static void Main()
        {
            double nota1 = double.Parse(Console.ReadLine());

            while (nota1 < 0 || nota1 > 10)
            {
                Console.WriteLine("nota invalida");
                nota1 = double.Parse(Console.ReadLine());
            }

            double nota2 = double.Parse(Console.ReadLine());

            while (nota2 < 0 || nota2 > 10)
            {
                Console.WriteLine("nota invalida");
                nota2 = double.Parse(Console.ReadLine());
            }

            Console.WriteLine($"media = {(nota1 + nota2) / 2.0:F2}");
        }
    }
}